src/v2/Apps/Shows/Components/ShowsHeader.tsx

```diff
@@ -0,0 +1,139 @@
+import {
+  Box,
+  BorderBox,
+  Text,
+  Join,
+  SelectSmall,
+  GridColumns,
+  Column,
+  space,
+  color,
+} from "@artsy/palette"
+import { graphql } from "lib/graphql"
+import React from "react"
+import { createFragmentContainer } from "react-relay"
+import styled from "styled-components"
+import { RouterLink } from "v2/Artsy/Router/RouterLink"
+import { useRouter } from "v2/Artsy/Router/useRouter"
+import { ShowsHeader_viewer } from "v2/__generated__/ShowsHeader_viewer.graphql"
+
+const City = styled(RouterLink)`
+  &.active {
+    color: ${color("purple100")};
+  }
+`
+
+// TODO: Extract this pattern
+const Overlay = styled(Box)<{ atEnd: boolean }>`
+  position: relative;
+  /* Fade-out gradient */
+  &::after {
+    display: block;
+    content: "";
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 2px;
+    width: ${space(6)}px;
+    z-index: 1;
+    pointer-events: none;
+    background: linear-gradient(
+      to right,
+      rgba(255, 255, 255, 0) 0%,
+      rgba(255, 255, 255, 1) 100%
+    );
+    /* Hide when scrolled all the way over */
+    transition: opacity 250ms;
+    opacity: ${({ atEnd }) => (atEnd ? 0 : 1)};
+  }
+`
+
+const Viewport = styled(Box)`
+  overflow-x: auto;
+`
+
+const Rail = styled(Box)`
+  display: inline-flex;
+  white-space: nowrap;
+  min-width: 100%;
+`
+
+interface ShowsHeaderProps {
+  viewer: ShowsHeader_viewer
+}
+
+const DEFAULT_CITIES = [
+  { text: "All", value: "" },
+  // { text: "Online Exclusive", value: "online" }, // TODO
```

Comment:
@anandaroop how bad of an idea do you think it is to resolve a fictional `online` city in the Metaphysics `CityType`

![keyboard_surfing_the_internet2-406x450](https://user-images.githubusercontent.com/112297/106330878-05dad580-6252-11eb-8808-9ace5be932f6.jpg)

Reply:
Whoa, `cyberspace`… 

It feels awkward to me, and I'm wondering if there's a legit user need for it. But what would happen here — a listing page for all online-exclusive shows? Maybe that could be better handled with its own landing page?

Reply:
We effectively treat "online" as a city in the frontend: https://www.artsy.net/shows/online

In general I think it makes sense to push any awkwardness into the graph; but it's weird in this case; admittedly. The alternative is to build out the args for the top-level shows connection in Metaphysics then create a copy of the city page with different queries.

Reply:
Ah, I see. It will be odd that `online` resolves to a City that has no coordinates, but weirdly the schema seems to allow that. 

Just to confirm this will purely be a matter of Force and MP agreeing on this `online` convention, no support needed in Gravity, yeah?

If so I can get with this (barring a re-think to how we handle online shows on the front end to begin with).

Reply:
Yeah; just resolving a little object with a slug/name for online then hardcoding an exception for it's showsConnection. 

But, I'm going to PR some additions to the top-level showsConnection anyway, because might as well see how awkward it really is (maybe it's not so bad 🤞 ).

Reply:
Hah... This 'online' hack just keeps on giving!

Reply:
It's weird to have an `online` "city" but less weird to call the same value "location" (if we could accomplish that across both Force and MP logic).